packages/progress-indicator/src/LionProgressIndicator.js

```diff
@@ -78,12 +126,42 @@ export class LionProgressIndicator extends LocalizeMixin(LitElement) {
 
   connectedCallback() {
     super.connectedCallback();
-    this.setAttribute('role', 'status');
+    const uid = uuid();
+    if (!this.hasAttribute('value')) {
+      this.setAttribute('role', 'status');
+    } else {
+      this.setAttribute('role', 'progressbar');
+      this.setAttribute('aria-valuenow', this.value.toString());
+      this.setAttribute('aria-valuemin', this.min.toString());
+      this.setAttribute('aria-valuemax', this.max.toString());
+      this.setAttribute('aria-label', this.label);
+    }
+    this.setAttribute('id', `progress-indicator-${uid}`);
     this.setAttribute('aria-live', 'polite');
   }
 
+  /**
+   * Update aria labels on state change.
+   * @param {import('@lion/core').PropertyValues } changedProperties
+   */
+  updated(changedProperties) {
+    if (changedProperties.has('value') && this.hasAttribute('value')) {
+      this.setAttribute('aria-valuenow', this.value.toString());
+    }
+  }
+
   onLocaleUpdated() {
-    const label = localize.msg('lion-progress-indicator:loading');
-    this.setAttribute('aria-label', label);
+    if (!this.hasAttribute('value') && !this.label) {
+      this.setAttribute('aria-label', localize.msg('lion-progress-indicator:loading'));
+    }
+  }
+
+  get _customStyles() {
```

Comment:
this could be added in a demo component since this is an unopinionated base component